### ISSUE TITLE
Add Group Entry as a separate class from LayerState

### ISSUE THREAD

**Issue by yileifeng** [RCSideProjects/RAMP-Vue-Testing#38]
Currently, there are a lot of edge cases that needs to be taken care of when toggling visibility and becomes excessive (especially in the case of nested group entries as elements of a visibility set). It is probably a good idea to make Group Entry (similar to Visibility Sets) as its own class to more take care of these cases.

**Comment by yileifeng** [comment]
Solution should also solve https://github.com/RCSideProjects/RAMP-Vue-Testing/issues/36 along with any additional bugs with nested visibility sets found along the way (there are a bunch)

**Comment by RyanCoulsonCA** [comment]
Yeah, this definitely makes sense. Group Entries aren't even layers themselves, so it doesn't really make sense for them to use the LayerState class.
